internal/worker/json.go

```diff
@@ -40,6 +40,9 @@ func (j OSBuildJob) OsbuildExports() []string {
 }
 
 type JobResult struct {
+	// build job results have the `arch` field
+	// use this for the job metrics arch
+	Arch     string              `json:"arch,omitempty"`
```

Comment:
Where is this set? Am I blind? :sweat_smile:

Reply:
@thozza added it here:
https://github.com/osbuild/osbuild-composer/blob/a5a16113355ea167376ddfb020ab50fac7e3bf8d/internal/worker/json.go#L55

So only the build jobs will have the `arch` field the rest won't

Reply:
I could just use an if/else statement and check the `jobType` instead of this though if that's more readable

Reply:
Aren't these fields stored separately, though? See https://go.dev/play/p/cz0tGv4iCar

Reply:
I vote for checking the job type and casting its results explicitly. I find explicit better than implicit and this code is a bit magic and works correctly only in very specific situations.

Reply:
Yeah makes sense, it was being unmarshalled here:
https://github.com/osbuild/osbuild-composer/blob/a5a16113355ea167376ddfb020ab50fac7e3bf8d/internal/worker/server.go#L603

But I agree that it's a bit magic and not very explicit. Busy refactoring it now.